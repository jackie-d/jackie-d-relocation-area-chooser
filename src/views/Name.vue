<template>
  <div>
    <navbar v-bind:showMenu="false" />
    <mdb-container style="height: 100%">
      
      <section class="text-center">
          <h2 class="h1 pt-4">{{ $t('name.title') }}</h2>

          <p class="text-primary font-weight-bold mb-5 mt-5 px-3">{{ $t('name.desc') }}</p>
          <mdb-row class="justify-content-sm-center">

            <mdb-col sm="6">
              <mdb-card testimonial>
                <mdb-card-up gradient="rainy-ashville"></mdb-card-up>
                <mdb-card-avatar color="white" class="mx-auto mt-4"><img :src="require('@/assets/avatar.jpg')" class="rounded-circle"></mdb-card-avatar>
                <mdb-card-body>
                  <mdb-card-title>{{ $t('name.your_name') }}</mdb-card-title>
                  <mdb-input 
                    v-model="name" 
                    ref="name"
                  />
                  <hr/>
                  <mdb-btn color="primary" v-on:click="goToCities" :disabled="name == undefined || name == ''">
                    <mdb-icon icon="play" />
                    {{ $t('name.next') }}
                  </mdb-btn>
                </mdb-card-body>
              </mdb-card>
            </mdb-col>

          </mdb-row>
        </section>
    
        <mdb-modal :show="modal" @close="modal = false">
          <mdb-modal-header>
            <mdb-modal-title>{{ $t('name.invalid_name_title') }}</mdb-modal-title>
          </mdb-modal-header>
          <mdb-modal-body>{{ $t('name.invalid_name_desc') }}</mdb-modal-body>
          <mdb-modal-footer>
            <mdb-btn color="secondary" @click.native="modal = false">{{ $t('common.close') }}</mdb-btn>
          </mdb-modal-footer>
        </mdb-modal>


    </mdb-container>
  </div>
</template>

<script lang="ts">
import Vue from 'vue';

import { mdbModal, mdbModalHeader, mdbModalTitle, mdbModalBody, mdbModalFooter, mdbContainer, mdbCard, mdbBtn, mdbRow, mdbCol, mdbCardUp, mdbCardAvatar, mdbCardBody, mdbCardTitle, mdbIcon, mdbInput } from 'mdbvue';
import navbar from '@/components/navbar.vue';
import mainFooter from '@/components/main-footer.vue';
import router from '../router'
import store from '../store';

export default Vue.extend({
  name: 'Home',
  data: function() {
    return {
      name: '',
      modal: false
    }
  },
  mounted() {
    this.$refs.name.$el.focus();
  },
  methods: {
    goToCities: function() {
        if ( this.name && this.name != '' ) {
          store.commit('SET_NAME', this.name);
          router.push('/cities');
        } else {
          this.modal = true;
        }
    }
  },
  components: {
    mdbContainer,
    mdbCard, 
    mdbBtn,
    navbar,
    mainFooter,
    mdbRow,
    mdbCol,
    mdbCardUp, 
    mdbCardAvatar, 
    mdbCardBody, 
    mdbCardTitle, 
    mdbIcon,
    mdbInput,
    mdbModal, 
    mdbModalHeader, 
    mdbModalTitle,
    mdbModalBody, 
    mdbModalFooter
  },
});
</script>