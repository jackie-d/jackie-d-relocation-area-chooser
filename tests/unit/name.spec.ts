import { shallowMount } from '@vue/test-utils'
import Name from '@/views/Name.vue';

describe('Name view', () => {
  it('Renders title', () => {
    const wrapper = shallowMount(Name, {
      mocks: {
        $t: (key) => key
      }
    });
    expect(wrapper.find('h2').text()).toMatch('name.title')
  })

  it('Snapshot consistence', () => {
    const wrapper = shallowMount(Name, {
        mocks: {
          $t: (key) => key
        }
      });
      expect(wrapper).toMatchSnapshot();
  });
  

})