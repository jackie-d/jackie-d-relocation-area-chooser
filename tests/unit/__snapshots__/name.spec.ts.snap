// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Name view Snapshot consistence 1`] = `
<div>
  <navbar-stub></navbar-stub>
  <mdb-container-stub tag="div" style="height: 100%;">
    <section class="text-center">
      <h2 class="h1 pt-4">name.title</h2>
      <p class="grey-text mb-5 mt-5 px-3">name.desc</p>
      <mdb-row-stub tag="div" class="justify-content-sm-center">
        <mdb-col-stub tag="div" sm="6">
          <mdb-card-stub tag="div" testimonial="true">
            <mdb-card-up-stub tag="div" gradient="rainy-ashville"></mdb-card-up-stub>
            <mdb-card-avatar-stub tag="div" color="white" class="mx-auto mt-4"><img src="" class="rounded-circle"></mdb-card-avatar-stub>
            <mdb-card-body-stub tag="div">
              <mdb-card-title-stub tag="h4">name.your_name</mdb-card-title-stub>
              <mdb-input-stub tag="input" type="text" value=""></mdb-input-stub>
              <hr>
              <mdb-btn-stub waves="true" tag="button" color="primary" type="button" disabled="true">
                <mdb-icon-stub icon="eye" color=""></mdb-icon-stub>
                Let's find the city
              </mdb-btn-stub>
            </mdb-card-body-stub>
          </mdb-card-stub>
        </mdb-col-stub>
      </mdb-row-stub>
    </section>
    <mdb-modal-stub tag="div" bgsrc="" direction="top">
      <mdb-modal-header-stub tag="div" close="true">
        <mdb-modal-title-stub tag="h5">name.invalid_name_title</mdb-modal-title-stub>
      </mdb-modal-header-stub>
      <mdb-modal-body-stub tag="div">name.invalid_name_desc</mdb-modal-body-stub>
      <mdb-modal-footer-stub tag="div">
        <mdb-btn-stub waves="true" tag="button" color="secondary" type="button">common.close</mdb-btn-stub>
      </mdb-modal-footer-stub>
    </mdb-modal-stub>
  </mdb-container-stub>
</div>
`;
